generate.py

```diff
@@ -0,0 +1,105 @@
+from collections import OrderedDict
+import json
+import os
+from sys import argv
+from bs4 import BeautifulSoup
+from deep_translator import GoogleTranslator
+
+
+def extract_i18n_keys(html_content):
+    soup = BeautifulSoup(html_content, 'html.parser')
+    i18n_dict = {}
+    for tag in soup.find_all(attrs={"data-i18n": True}):
+        i18n_values = str(tag.attrs.get("data-i18n")).split(";")
+
+        for value in i18n_values:
+            if value.startswith('['):
+                end_bracket_pos = value.find(']')
+                if end_bracket_pos != -1:
+                    attribute_name = value[1:end_bracket_pos]
+                    key = value[end_bracket_pos+1:]
+                    value = tag.attrs.get(attribute_name, '')
+                    i18n_dict[key] = value
+            else:
+                key = value
+                value = tag.text.strip()
+                i18n_dict[key] = value
+    return i18n_dict
+
+
+def process_html_files(directory):
+    i18n_data = {}
+    for root, dirs, files in os.walk(directory):
+        for file in files:
+            if file.endswith('.html'):
+                filepath = os.path.join(root, file)
+                with open(filepath, 'r', encoding='utf-8') as f:
+                    html_content = f.read()
+
+                i18n_data.update(extract_i18n_keys(html_content))
+    return i18n_data
+
+
+def update_json(json_file, i18n_dict):
+    with open(json_file, 'r', encoding='utf-8') as file:
+        data = json.load(file, object_pairs_hook=OrderedDict)
+
+    try:
+        for key in i18n_dict.keys():
+            if key not in data and i18n_dict[key] != "":
+                print(f"Key '{key}' not found in '{json_file}'.")
+                data[key] = GoogleTranslator(source='en', target='zh-CN').translate(i18n_dict[key])
```

Comment:
Target is hard coded?

Reply:
fixed at [`65eb528` (#1)](https://github.com/SillyTavern/SillyTavern-i18n/pull/1/commits/65eb5282580034fdc2be7bc0d49f277ea7132c37).